src/coroutine.rs

```diff
@@ -35,6 +35,10 @@ pub struct Coroutine {
     waker: Option<Arc<AsyncioWaker>>,
 }
 
+// Safety: `Coroutine` is allowed to be `Sync` even though the future is not,
+// because the future is polled with `&mut self` receiver
+unsafe impl Sync for Coroutine {}
```

Comment:
I marked this ready for review, but I think this hasn't been looked at yet. I'll leave that to David since I'm not sure if using `Exclusive` makes sense.